lib/media/playhead.js

```diff
@@ -416,6 +442,30 @@ shaka.media.MediaSourcePlayhead = class {
     this.gapController_.onSegmentAppended();
   }
 
+  /** @override */
+  isBufferedToEnd() {
+    goog.asserts.assert(
+        this.mediaElement_,
+        'We need a video element to get buffering information');
+    goog.asserts.assert(
+        this.timeline_,
+        'We need a timeline to get buffering information');
+
+    // Live streams are "buffered to the end" when they have buffered to the
+    // live edge or beyond (into the region covered by the presentation delay).
+    if (this.timeline_.isLive()) {
+      const liveEdge = this.timeline_.getSegmentAvailabilityEnd();
+      const bufferEnd =
+          shaka.media.TimeRangesUtils.bufferEnd(this.mediaElement_.buffered);
+
+      if (bufferEnd != null && bufferEnd >= liveEdge) {
+        return true;
+      }
+    }
+
+    return false;
```

Comment:
This never returns true for VOD.  I think this might be the source of the problem.

This case from the original code isn't represented anywhere:

```
    // This means that MediaSource has buffered the final segment in all
    // SourceBuffers and is no longer accepting additional segments.
    if (this.mediaSourceEngine_.ended()) {
      return true;
    }
```